From: Francois Romieu <romieu@fr.zoreil.com>
Date: Wed, 27 Apr 2011 13:52:22 -0700
Subject: [PATCH 57/71] r8169: fix merge conflict fix.

commit 15ecd039b7182d725f4294e01f2fb12c3a88db17 upstream.

- use adequate MAC_VER id
  (see 01dc7fec4025f6bb72b6b98ec88b375346b6dbbb)
- remove duplicate rtl_firmware_info record
- remove duplicate functions

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Realtek linux nic maintainers <nic_swsd@realtek.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Drop inapplicable changes]
---
 drivers/net/r8169.c |   26 ++++++--------------------
 1 files changed, 6 insertions(+), 20 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 128b80a..64c5ea0 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -190,7 +190,9 @@ static const struct rtl_firmware_info {
 	{ .mac_version = RTL_GIGA_MAC_VER_25, .fw_name = FIRMWARE_8168D_1 },
 	{ .mac_version = RTL_GIGA_MAC_VER_26, .fw_name = FIRMWARE_8168D_2 },
 	{ .mac_version = RTL_GIGA_MAC_VER_29, .fw_name = FIRMWARE_8105E_1 },
-	{ .mac_version = RTL_GIGA_MAC_VER_30, .fw_name = FIRMWARE_8105E_1 }
+	{ .mac_version = RTL_GIGA_MAC_VER_30, .fw_name = FIRMWARE_8105E_1 },
+	{ .mac_version = RTL_GIGA_MAC_VER_32, .fw_name = FIRMWARE_8168E_1 },
+	{ .mac_version = RTL_GIGA_MAC_VER_33, .fw_name = FIRMWARE_8168E_2 }
 };
 
 enum cfg_version {
@@ -2592,6 +2594,8 @@ static void rtl8168e_hw_phy_config(struct rtl8169_private *tp)
 		{ 0x1f, 0x0000 }
 	};
 
+	rtl_apply_firmware(tp);
+
 	rtl_writephy_batch(tp, phy_reg_init, ARRAY_SIZE(phy_reg_init));
 
 	/* DCO enable for 10M IDLE Power */
@@ -2633,22 +2637,6 @@ static void rtl8168e_hw_phy_config(struct rtl8169_private *tp)
 	rtl_writephy(tp, 0x0d, 0x0000);
 }
 
-static void rtl8168e_1_hw_phy_config(struct rtl8169_private *tp)
-{
-	if (rtl_apply_firmware(tp, FIRMWARE_8168E_1) < 0)
-		netif_warn(tp, probe, tp->dev, "unable to apply firmware patch\n");
-
-	rtl8168e_hw_phy_config(tp);
-}
-
-static void rtl8168e_2_hw_phy_config(struct rtl8169_private *tp)
-{
-	if (rtl_apply_firmware(tp, FIRMWARE_8168E_2) < 0)
-		netif_warn(tp, probe, tp->dev, "unable to apply firmware patch\n");
-
-	rtl8168e_hw_phy_config(tp);
-}
-
 static void rtl8102e_hw_phy_config(struct rtl8169_private *tp)
 {
 	static const struct phy_reg phy_reg_init[] = {
@@ -2764,10 +2752,8 @@ static void rtl_hw_phy_config(struct net_device *dev)
 		rtl8105e_hw_phy_config(tp);
 		break;
 	case RTL_GIGA_MAC_VER_32:
-		rtl8168e_1_hw_phy_config(tp);
-		break;
 	case RTL_GIGA_MAC_VER_33:
-		rtl8168e_2_hw_phy_config(tp);
+		rtl8168e_hw_phy_config(tp);
 		break;
 
 	default:
-- 
1.7.5.4

